model.py

```diff
@@ -9,43 +9,98 @@
 import utils
 
 
+class GaussianAttention(nn.Module):
+    def __init__(self, query_size=5):
+        super(GaussianAttention, self).__init__()
+        self.linear = nn.Linear(in_features=query_size, out_features=2)
+
+    def forward(self, query, embedded_text, text_mask, position):
+        """query should be of shape [seq_len, batch_size, query_size]]
+        textEmbedding should be of shape [text_len, batch_size]
+        text_mask should be of shape [text_len, batch_size]
+        position should be of shape [seq_len, batch_size] (or None)
+        Returns tuple of:
+          * a tensor of shape [seq_len, batch_size, embedding_dim]
+          * a tensor of shape [seq_len, batch_size] containing positions."""
+        # TODO(neil): rewrite so that we don't need seq_len leading dimension
+        masked_embedding = embedded_text * text_mask.unsqueeze(-1)
+        seq_len = query.shape[0]
+        batch_size = query.shape[1]
+        if position is None:
+            position = torch.zeros((seq_len, batch_size), device=query.device)
```

Comment:
I'm suspicious about this line. In particular, why would this have seq_len as one of the dimensions?

Reply:
I think that you don't observe a bug because you either call this with `seq_len==1` or with `position = None`, however, this looks wrong in general.